app/controllers/TaxYearSummaryController.scala

```diff
@@ -228,20 +234,29 @@ class TaxYearSummaryController @Inject()(taxYearSummaryView: TaxYearSummary,
   private def handleRequest(taxYear: Int, origin: Option[String], isAgent: Boolean)
                            (implicit user: MtdItUser[AnyContent], hc: HeaderCarrier,
                             ec: ExecutionContext, messages: Messages): Future[Result] = {
+
     withTaxYearFinancials(taxYear, isAgent) { charges =>
       withObligationsModel(taxYear, isAgent) { obligationsModel =>
         val codingOutEnabled: Boolean = isEnabled(CodingOut)
         val mtdItId: String = if (isAgent) getClientMtditid else user.mtditid
         val nino: String = if (isAgent) getClientNino else user.nino
-        calculationService.getLiabilityCalculationDetail(mtdItId, nino, taxYear).map { liabilityCalcResponse =>
-          view(liabilityCalcResponse, charges, taxYear, obligationsModel, codingOutEnabled,
+        for {
+          viewModel <- claimToAdjustViewModel(Nino(value = user.nino), taxYear)
```

Comment:
not sure if this is a correct order, there are quite a few cases where we don't need to call claimToAdjustViewModel(Nino(value = user.nino), taxYear) based on liabilityCalcResponse

Reply:
I'm not really familiar with liabilityCalcResponse. For what conditions would we not need to check?

Reply:
have a look at the current implementation for the next method:
TaxYearSummaryController =>  private def view( ...
in some cases we really do not need to claimToAdjustViewModel

Reply:
I didn't want to make claimToAdjustViewModel optional as this moves more logic to the view. So when we don't have anything to show on the page we just put in the viewmodel with nothing in it, like 
`      Future.successful(TYSClaimToAdjustViewModel(isEnabled(AdjustPaymentsOnAccount), None))
`
Maybe this is not the best approach?